plugins/modules/mssql_script.py

```diff
@@ -275,6 +297,10 @@ def run_module():
 
     conn.autocommit(True)
 
+    # If transactional mode is requested, start a transaction
+    if transaction:
+        conn.autocommit(False)
```

Comment:
How about simplifying this a bit:
```suggestion
    # If transactional mode is requested, start a transaction
    conn.autocommit(not transaction)
```